### ISSUE TITLE
Support SecretBinding for DNS Providers

### ISSUE THREAD

**Issue by timuthy** [gardener/gardener#3846]
**How to categorize this issue?**

/area usability
/kind enhancement
/priority 3

**What would you like to be added**:
The `Shoot` API should be extended by an optional `secretBindingName` field which can be used as an alternative for `shoot.spec.dns.providers[].secretName` to set up DNS providers:

```yaml
spec:
  dns:
    domain: foo.bar
    providers:
      - secretBindingName: gcp-secret-binding
        type: google-clouddns
      - secretName: aws-secret
        type: aws-route53
```

**Why is this needed**:
Today, DNS providers for shoot clusters refer to `Secrets` that reside in the very same namespace as the `Shoot`.
In addition, users might want to refer to `Secrets` in other project namespaces where they have access to.

/cc @grolu

**Comment by rfranzke** [comment]
@timuthy Does this request still make sense in the light of #5270?

**Comment by timuthy** [comment]
It's still a valid use case but we need to think about it again as with #5270 we would need a new approach to tackle this. Let's close this for now and discuss again once we implemented #5270.
/close

**Comment by grolu** [comment]
I just want to make clear that it is not only about adding the possibility to reference secrets from another namespace. It is also behaving differently than the shoot secrets, therefore the Dashboard secret handling is kind of hacky atm. We need to create a secret binding resource even if we cannot use them in the spec, so users could be very confused why a secret binding resource is required for DNS secrets to show up in the Dashboard. The main reason is that the secret binding resource is the place where we store the provider type information. I don't want to start a discussion about this now, however please consider adding support for secret binding resources when you refactor this feature because this would really help us keeping the Dashboard code cleaner.
